app/src/main/java/com/android/sample/ui/mainscreen/ViewAllScreen.kt

```diff
@@ -0,0 +1,236 @@
+package com.android.sample.ui.mainscreen
+
+import androidx.annotation.StringRes
+import androidx.compose.foundation.background
+import androidx.compose.foundation.border
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.PaddingValues
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material3.*
+import androidx.compose.material3.Card
+import androidx.compose.material3.CardDefaults
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Scaffold
+import androidx.compose.material3.Text
+import androidx.compose.runtime.*
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import com.android.sample.R
+import com.android.sample.model.workout.BodyWeightWorkout
+import com.android.sample.model.workout.Workout
+import com.android.sample.model.workout.WorkoutViewModel
+import com.android.sample.model.workout.YogaWorkout
+import com.android.sample.ui.composables.ImageComposable
+import com.android.sample.ui.composables.TopBar
+import com.android.sample.ui.navigation.NavigationActions
+import com.android.sample.ui.theme.Blue
+
+/**
+ * Displays the screen for viewing all workouts.
+ *
+ * @param navigationActions The navigation actions to handle navigation events.
+ * @param bodyWeightViewModel The ViewModel for managing bodyweight workouts.
+ * @param yogaViewModel The ViewModel for managing yoga workouts.
+ */
+@Composable
+fun ViewAllScreen(
+    navigationActions: NavigationActions,
+    bodyWeightViewModel: WorkoutViewModel<BodyWeightWorkout>,
+    yogaViewModel: WorkoutViewModel<YogaWorkout>,
+    // runningViewModel: WorkoutViewModel<Workout>
+) {
+  val profile = R.drawable.homme
+  var selectedTab by remember { mutableIntStateOf(0) } // (0 = Bodyweight, 1 = Yoga, 2 = Running)
+
```

Comment:
I thinks there is a better way to map 0 -> Bodywight 1->Yoga and 2 -> Running. But I think this is OK as you commented . Using a Const value for each would be unnecessary.